src/Disks/ObjectStorages/IMetadataStorage.h

```diff
@@ -183,6 +183,11 @@ class IMetadataStorage : private boost::noncopyable
 
     virtual MetadataStorageType getType() const = 0;
 
+    /// Returns true if empty file can be created without any blobs in the corresponding object storage.
+    /// E.g. metadata storage can store the empty list of blobs corresponding to a file without actually storing any blobs.
+    /// But if the metadata storage just relies on for example local FS to store data under logical path, then a file has to be created even if it's empty.
+    virtual bool supportsEmptyFilesWithoutBlobs() const { return false; }
```

Comment:
This makes sense, so the change will not affect the system tables in the cloud.

Reply:
Yep, let's enable this optimization explicitly for specific disks types